editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -5767,8 +5766,8 @@ void CanvasItemEditorViewport::_create_nodes(Node *parent, Node *child, String &
 		undo_redo->add_undo_method(ed, "live_debug_remove_node", NodePath(String(EditorNode::get_singleton()->get_edited_scene()->get_path_to(parent)) + "/" + new_name));
 	}
 
-	if (Object::cast_to<TouchScreenButton>(child) || Object::cast_to<TextureButton>(child)) {
-		undo_redo->add_do_property(child, "texture_normal", texture);
+	if (child->has_method("set_texture_normal")) {
+		undo_redo->add_do_method(child, "set_texture_normal", texture);
```

Comment:
What's the rationale for this change?

Reply:
Original motivation: This avoids a hard dependency on `TouchScreenButton` which is in fact not a `Button` in Godot but instead inherits `Node2D`, so it can't be used if compiling without `Node2D`.

Intent: Instead of looking for classes that have `texture_normal`, let's just look for `texture_normal`.

However, since there is no `has_property`, I had to use `has_method` instead. We should consider adding `has_property` though, that would be better I think.

Reply:
There's another hard dependency on `TouchScreenButton` in this class though, so it doesn't really solve anything as is:
```cpp
        // Adjust position for Control and TouchScreenButton
        if (Object::cast_to<Control>(child) || Object::cast_to<TouchScreenButton>(child)) {
                target_position -= texture->get_size() / 2;
        }
```

Reply:
True, this is non-trivial. I'll remove for this PR.